src/openloco/industry.cpp

```diff
@@ -119,69 +119,82 @@ namespace openloco
         if (!(flags & industry_flags::flag_01) && under_construction == 0xFF)
         {
             // Run tile loop for 100 iterations
-            auto obj = object();
             for (int i = 0; i < 100; i++)
             {
-                const auto& surface = tilemgr::get(tile_loop.current()).surface();
-                if (surface != nullptr)
+                sub_45329B(tile_loop.current());
+                if (tile_loop.next() == map_pos())
                 {
-                    if (surface->data()[0] & 0x80)
+                    sub_453354();
+                    break;
+                }
+            }
+        }
+    }
+
+    // 0x0045329B
+    void industry::sub_45329B(const map_pos& pos)
+    {
+        const auto& surface = tilemgr::get(pos).surface();
+        if (surface != nullptr)
+        {
+            if (surface->has_high_type_flag())
+            {
+                if (surface->industry_id() == id())
+                {
+                    uint8_t bl = surface->var_6_SLR5();
+                    auto obj = object();
+                    if (bl == 0 || bl != obj->var_EA)
                     {
-                        if (surface->industry_id() == id())
+                        var_DB++;
+                        if ((!(obj->flags & industry_object_flags::flag_28) && surface->var_4_E0() == 0) || find_5(surface))
                         {
-                            uint8_t bl = surface->data()[6] >> 5;
-                            if (bl == 0 || bl != obj->var_EA)
-                            {
-                                var_DB++;
-                                if ((!(obj->flags & industry_object_flags::flag_29) && (surface->data()[4] & 0xE0) == 0) || find_5(surface))
-                                {
-                                    var_DD++;
-                                }
-                            }
+                            var_DD++;
                         }
                     }
                 }
-                if (tile_loop.next() == map_pos())
-                {
-                    int16_t tmp_a = var_DB / 16;
-                    int16_t tmp_b = std::max(0, var_DD - tmp_a);
-                    int16_t tmp_c = var_DB - tmp_b;
-                    int16_t tmp_d = std::min(tmp_c / 25, 255);
-                    if (tmp_d < obj->var_EB)
-                    {
-                        var_DF = tmp_d / obj->var_EB;
-                    }
-                    else
-                    {
-                        var_DF = 255;
-                    }
+            }
+        }
+    }
+
+    void industry::sub_453354()
+    {
+        auto obj = object();
+        int16_t tmp_a = var_DB / 16;
```

Comment:
if we are refactoring might as well move this to `camelCase`

Reply:
Not necessary for this PR, I think.